### PR TITLE
[#862] Add ZK option to Institution

### PR DESCRIPTION
**Pull request by jswk**
Add a boolean field zk to Institution and propagate it in the hierarchy.

The field should be settable by admin in AdminInstitutionController.
Allow only to unset it on the root of ZK (that is on an institution
which has no parent or its parent has zk==false).

Use it to populate ROLE_ZK_MEMBER.
Remove obsolete AppUser.zkMember.

Closes: #862.

### CODE REVIEW

File: s4e-backend/src/main/java/pl/cyfronet/s4e/admin/institution/AdminInstitutionController.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            @ApiResponse(responseCode = "403", description = "Forbidden", content = @Content)
+    })
+    @PatchMapping(path = "/{slug}/zk/set")
+    public void setZk(@PathVariable String slug) throws NotFoundException, InstitutionZkException {
+        institutionService.setZk(slug);
+    }
+
+    @Operation(summary = "Unset ZK flag")
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "OK"),
```
Comment (danielkryska) [inline s4e-backend/src/main/java/pl/cyfronet/s4e/admin/institution/AdminInstitutionController.java:41]: Wouldn't be better to response with message like `zk flag was unset`?

File: s4e-backend/src/main/java/pl/cyfronet/s4e/admin/institution/AdminInstitutionController.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+@RestController
+@RequestMapping(path = ADMIN_PREFIX + "/institutions", produces = APPLICATION_JSON_VALUE)
+@Tag(name = "admin-institution", description = "The Admin Institution API")
+@RequiredArgsConstructor
+public class AdminInstitutionController {
+    private final InstitutionService institutionService;
+
+    @Operation(summary = "Set ZK flag")
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "OK"),
```
Comment (danielkryska) [inline s4e-backend/src/main/java/pl/cyfronet/s4e/admin/institution/AdminInstitutionController.java:29]: Wouldn't be better to response with message like `zk flag was set`?

File: s4e-backend/src/main/java/pl/cyfronet/s4e/service/InstitutionService.java [hunk 3]
```diff
@@ -29,65 +25,91 @@
...
 
     @Transactional(rollbackFor = InstitutionCreationException.class)
-    public Institution save(Institution institution) throws InstitutionCreationException {
-        try {
-            return institutionRepository.save(institution);
-        } catch (DataIntegrityViolationException e) {
-            log.info("Cannot create Institution with name '" + institution.getName() + "'", e);
-            throw new InstitutionCreationException("Cannot create Institution", e);
-        }
+    public String create(Institution institution) throws InstitutionCreationException {
```
Comment (danielkryska) [inline s4e-backend/src/main/java/pl/cyfronet/s4e/service/InstitutionService.java:44]: why this function have been renamed from `save` to `create`? Haven't used before `save` name of such methods?

File: s4e-backend/src/test/java/pl/cyfronet/s4e/TestJwtUtil.java [hunk 4]
```diff
@@ -68,9 +68,9 @@ private static AppUserDetails createAppUserDetails(AppUser user) {
         if (user.isAdmin()) {
             authorities.add(new SimpleGrantedAuthority("ROLE_ADMIN"));
         }
-        if (user.isMemberZK()) {
-            authorities.add(new SimpleGrantedAuthority("ROLE_MEMBER_ZK"));
-        }
+//        if (user.isMemberZK()) {
```
Comment (danielkryska) [inline s4e-backend/src/test/java/pl/cyfronet/s4e/TestJwtUtil.java:71]: Is the commented code necessary?

**Review by danielkryska — APPROVED** [review]

**Review by kmarszalek — APPROVED** [review]

File: s4e-backend/src/main/java/pl/cyfronet/s4e/service/InstitutionService.java (hunk 3 above)
Comment (jswk) [reply s4e-backend/src/main/java/pl/cyfronet/s4e/service/InstitutionService.java:44]: There are some methods with save, although most of the creational methods (among services) stem from create. Also, the child-creating method in InstService already had the create in name, so I homogenized the naming.

**Pull request merged**